### DISCUSSION TITLE
Is there a way to generate coverage reports from japa?

### DISCUSSION THREAD

**Discussion by eidellev** [adonisjs/core#1363] [Help] [answered]
I'm not sure if this the right place for this question, but it would be really great to have this feature.

**Comment by McSneaky** [comment, marked as answer]
You should be able to do it with [NYC ](https://www.npmjs.com/package/nyc)

https://forum.adonisjs.com/t/output-coverage-report-from-tests/927/10

- Install `nyc`
- Add nyc config file `.nycrc` to root of project
    ```
    {
      "all": true,
      "include": ["app/**"],
      "reporter": "html",
      "report-dir": "test/coverage"
    }
    ```
- Add `nyc` before your start command `nyc node ace test`

**Comment by thetutlage** [comment]
Yup, as pointed by @McSneaky. You can use nyc. To get started quickly, the following command should do the job

```sh
nyc node japaFile.js
```

**Comment by eidellev** [comment]
thanks @McSneaky  @thetutlage! that's just what I needed

**Comment by FeMaffezzolli** [comment]
I've just used nyc along with AdonisJS V5 (after this [setup](https://preview.adonisjs.com/blog/running-tests-in-adonisjs-v5#app)) and I can confirm it works like a charm.
